script.js

```diff
@@ -1,4 +1,135 @@
+    outputMessage = rollTwoDiceForPlayer();
+    gameState = GAME_STATE_CHOOSE_DICE_ORDER;
+    return outputMessage;
+  }
+  //Select dice order for Player 1
+  if (gameState === GAME_STATE_CHOOSE_DICE_ORDER && playerTurn === 1) {
+    console.log(`choosing dice order.....`);
+    // If player inputs 1
+    if (input === "1") {
+      var player1Score = Number(
+        String(player1Rolls[0]) + String(player1Rolls[1])
+      );
+      playerScores.push([player1Score]);
+      playerTurn = 2;
+      gameState = GAME_STATE_DICE_ROLL;
+      return `Your chosen value is ${player1Score}. Player 2, please click the submit button to roll your dice!`;
+    }
+    // If player inputs 2
+    if (input === "2") {
+      player1Score = Number(String(player1Rolls[1]) + String(player1Rolls[0]));
+      playerScores.push([player1Score]);
+      playerTurn = 2;
+      gameState = GAME_STATE_DICE_ROLL;
+      return `Your chosen value is ${player1Score}. Player 2, please click the submit button to roll your dice!`;
+    }
+    //Ensure input is valid
+    if (input !== "1" && input !== "2") {
+      return `Error! Please input 1 or 2!`;
+    }
+  }
+
+  // Roll Dice for Player 2
+  if (gameState === GAME_STATE_DICE_ROLL && playerTurn === 2) {
+    outputMessage = rollTwoDiceForPlayer();
+    gameState = GAME_STATE_CHOOSE_DICE_ORDER;
+    return outputMessage;
+  }
+  //Select dice order for Player 2
+  if (gameState === GAME_STATE_CHOOSE_DICE_ORDER && playerTurn === 2) {
+    console.log(`choosing dice order.....`);
+    // If player inputs 1
+    if (input === "1") {
+      var player2Score = Number(
+        String(player2Rolls[0]) + String(player2Rolls[1])
+      );
+      playerScores.push([player2Score]);
+      gameState = GAME_STATE_COMPARE_SCORES;
+      return `Your chosen value is ${player2Score}. Now click the submit button to check who wins!`;
+    }
+    // If player inputs 2
+    if (input === "2") {
+      player2Score = Number(String(player2Rolls[1]) + String(player2Rolls[0]));
+      playerScores.push([player2Score]);
+      gameState = GAME_STATE_COMPARE_SCORES;
+      return `Your chosen value is ${player2Score}. Now click the submit button to check who wins!`;
+    }
+    //Ensure input is valid
+    if (input !== "1" && input !== "2") {
+      return `Error! Please input 1 or 2!`;
+    }
+  }
+
+  //Compare Scores and output winner
+  if (gameState === GAME_STATE_COMPARE_SCORES) {
+    var outputMessage;
+    if (playerScores[0] === playerScores[1]) {
+      outputMessage = `WOW WHAT LUCK! Both Player 1 and 2 have the same score of ${playerScores[0]}!!!!!<br><br>Click Submit to play again!`;
+    }
+    if (playerScores[0] > playerScores[1]) {
+      gamestate = GAME_STATE_DICE_ROLL;
+      outputMessage = `Player 1 Wins with a score of ${playerScores[0]}! Player 2 got ${playerScores[1]}:(<br><br>Click Submit to play again!`;
+    }
+    if (playerScores[0] < playerScores[1]) {
```

Comment:
this is a good usecase of `else if` and `else` so the program doesnt unnecessarily continue checking!